velox/expression/ExprToSubfieldFilter.h

```diff
@@ -488,7 +492,7 @@ class ExprToSubfieldFilterParser {
 
  private:
   // Factory method to create a parser instance.
-  static std::function<std::unique_ptr<ExprToSubfieldFilterParser>()>
+  static std::function<std::shared_ptr<ExprToSubfieldFilterParser>()>
```

Comment:
static inline to avoid storage declaration in cc.

Reply:
It's not possible to initialize it on the same line because the subclass relation of `class PrestoExprToSubfieldFilterParser : public ExprToSubfieldFilterParser` has to be defined after that, so we cannot cast `shared_ptr<PrestoExprToSubfieldFilterParser>` to `shared_ptr<ExprToSubfieldFilterParser>` at this point